plugins/line-numbers/prism-line-numbers.js

```diff
@@ -1,38 +1,85 @@
-Prism.hooks.add('after-highlight', function (env) {
-	// works only for <code> wrapped inside <pre> (not inline)
-	var pre = env.element.parentNode;
-	var clsReg = /\s*\bline-numbers\b\s*/;
-	if (
-		!pre || !/pre/i.test(pre.nodeName) ||
-		// Abort only if nor the <pre> nor the <code> have the class
-		(!clsReg.test(pre.className) && !clsReg.test(env.element.className))
-	) {
-		return;
-	}
-
-	if (clsReg.test(env.element.className)) {
-		// Remove the class "line-numbers" from the <code>
-		env.element.className = env.element.className.replace(clsReg, '');
-	}
-	if (!clsReg.test(pre.className)) {
-		// Add the class "line-numbers" to the <pre>
-		pre.className += ' line-numbers';
-	}
-
-	var linesNum = (1 + env.code.split('\n').length);
-	var lineNumbersWrapper;
-
-	var lines = new Array(linesNum);
-	lines = lines.join('<span></span>');
-
-	lineNumbersWrapper = document.createElement('span');
-	lineNumbersWrapper.className = 'line-numbers-rows';
-	lineNumbersWrapper.innerHTML = lines;
-
-	if (pre.hasAttribute('data-start')) {
-		pre.style.counterReset = 'linenumber ' + (parseInt(pre.getAttribute('data-start'), 10) - 1);
-	}
-
-	env.element.appendChild(lineNumbersWrapper);
-
-});
\ No newline at end of file
+(function(){
+
+	/**
+	 * Class name which is used to flag this code field has break-word css attr
+	 * @type {String}
+	 */
+	var BREAK_WORD_CLASS = 'line-numbers-break-word';
+
+	/**
+	 * Resizes line numbers spans according to height of line of code
+	 * @param  {Element} element <code> element
+	 */
+	var _resizeElement = function(element){
+		var lineNumbersWrapper = element.querySelector('.line-numbers-rows');
+		var lineNumberSizer = element.querySelector('.line-numbers-sizer');
+		var codeLines = element.textContent.split('\n');
+		var lineHeight;
+
+		lineNumberSizer.style.display = 'block';
+
+		// parse line height in such way because of problems with zoom & line height css attr
+		lineNumberSizer.innerText = '\n';
+		lineHeight = lineNumberSizer.getBoundingClientRect().height;
+
+		codeLines.forEach(function(line, lineNumber){
+			lineNumberSizer.innerText = line;
+			var lineSize = lineNumberSizer.getBoundingClientRect().height || lineHeight;
```

Comment:
Oh, probably it was legacy from old scenario, and can be deleted. 
in forEach block empty line height calculating with same scenario.

I can update the pull request with removing this lines (20-23). 

Everything else?